rust/src/lib/ovsdb/db.rs

```diff
@@ -45,98 +33,84 @@ impl OvsDbCondition {
+        self.transaction_id
+    }
+
     // TODO: support environment variable OVS_DB_UNIX_SOCKET_PATH
     pub(crate) fn new() -> Result<Self, NmstateError> {
         Ok(Self {
             rpc: OvsDbJsonRpc::connect(DEFAULT_OVS_DB_SOCKET_PATH)?,
+            transaction_id: get_sec_since_epoch(),
         })
     }
 
     pub(crate) fn check_connection(&mut self) -> bool {
-        if let Ok(reply) = self.rpc.exec("list_dbs", &Value::Array(vec![])) {
-            if let Some(dbs) = reply.as_array() {
-                dbs.iter().any(|db| db.as_str() == Some(OVS_DB_NAME))
-            } else {
-                false
-            }
+        let transaction_id = self.get_transaction_id();
+        let value = OvsDbMethodEcho::to_value(transaction_id);
+        if self.rpc.send(&value).is_ok() {
+            self.rpc.recv(transaction_id).is_ok()
         } else {
             false
         }
     }
 
+    pub(crate) fn transact(
+        &mut self,
+        transact: &OvsDbMethodTransact,
+    ) -> Result<Value, NmstateError> {
+        let transaction_id = self.get_transaction_id();
+        let value = transact.to_value(transaction_id);
+        self.rpc.send(&value)?;
+        let reply = self.rpc.recv(transaction_id)?;
+        check_transact_error(reply)
+    }
+
     fn _get_ovs_entry(
         &mut self,
         table_name: &str,
         columns: Vec<&'static str>,
     ) -> Result<HashMap<String, OvsDbEntry>, NmstateError> {
-        let select = OvsDbSelect {
-            table: table_name.to_string(),
-            conditions: vec![],
-            columns: Some(columns),
-        };
+        let reply = self.transact(&OvsDbMethodTransact {
+            db_name: OVS_DB_NAME.to_string(),
+            operations: vec![OvsDbOperation::Select(OvsDbSelect {
+                table: table_name.to_string(),
+                conditions: vec![],
+                columns: Some(columns),
+            })],
+        })?;
+
         let mut ret: HashMap<String, OvsDbEntry> = HashMap::new();
-        match self.rpc.exec(
-            "transact",
-            &Value::Array(vec![
-                Value::String(OVS_DB_NAME.to_string()),
-                select.to_value(),
-            ]),
-        )? {
-            Value::Array(reply) => {
-                if let Some(entries) = reply
-                    .first()
-                    .and_then(|v| v.as_object())
-                    .and_then(|v| v.get("rows"))
-                    .and_then(|v| v.as_array())
-                {
-                    for entry in entries {
-                        let ovsdb_entry: OvsDbEntry = entry.try_into()?;
-                        if !ovsdb_entry.uuid.is_empty() {
-                            ret.insert(
-                                ovsdb_entry.uuid.to_string(),
-                                ovsdb_entry,
-                            );
-                        }
+
+        if let Value::Array(reply) = reply {
+            if let Some(entries) = reply
```

Comment:
I think that this can be unified to `if let Value::Array(Some(entries)) = reply` (both `else` blocks are identical).

Reply:
They are not identical, the `reply` is different.

Reply:
It is `serde_json::Value(Vec<reply>)`, not `serde_json::Value(Option<Vec<reply>>)`.

Reply:
Never mind. I noticed `serde_json::Value::as_array()` can be use in chain actions.

Reply:
Changed to
```rust
        if let Some(entries) = reply
            .as_array()
            .and_then(|reply| reply.first())
            .and_then(|v| v.as_object())
            .and_then(|v| v.get("rows"))
            .and_then(|v| v.as_array())
        {
```

Reply:
Didn't look very carefully to this. Indeed, you cannot do what I said in this case. The modification is nice, though, thanks.